common/pango.c

```diff
@@ -88,13 +84,15 @@ PangoLayout *get_pango_layout(cairo_t *cairo, const char *font,
 
 void get_text_size(cairo_t *cairo, const char *font, int *width, int *height,
 		int *baseline, double scale, bool markup, const char *fmt, ...) {
-	char buf[MAX_CHARS];
-
 	va_list args;
 	va_start(args, fmt);
-	if (vsnprintf(buf, sizeof(buf), fmt, args) >= MAX_CHARS) {
-		strcpy(&buf[sizeof(buf) - sizeof(overflow)], overflow);
-	}
+	// Add one since vsnprintf excludes null terminator.
+	int length = vsnprintf(NULL, 0, fmt, args) + 1;
```

Comment:
Nitpick: I prefer to use "length" (or "len") to mean the return value of `strlen` (aka without the terminating NULL byte) and "size" to mean the value passed to `malloc` et al (aka including the terminating NULL byte).

Reply:
Yep, that's fair. I can change it if you really want.